server/src/main/java/com/defold/extender/services/DefoldSdkService.java

```diff
@@ -148,81 +154,74 @@ public File getRemoteSdk(String hash) throws IOException, URISyntaxException, Ex
-                while (Files.exists(lockFile.toPath()) && seconds > 0) {
-                    int step = 2;
-                    seconds -= step;
-                    try {
-                        Thread.sleep(step * 1000);
-                    } catch (InterruptedException e) {
-                        // pass
+                    } catch (IOException|URISyntaxException exc) {
+                        LOGGER.error("Error downloading defoldsdk", exc);
                     }
                 }
 
-                // Now check again that the SDK was downloaded
-                if (!Files.exists(sdkDirectory.toPath())) {
-                    throw new ExtenderException(String.format("The given sdk still does not exist: %s", hash));
+                if (!sdkFound) {
+                    sdk.close();
+                    return null;
                 }
-            }
-        }
 
-        LOGGER.info("Using Defold SDK version {}", hash);
-
-        MetricsWriter.metricsTimer(meterRegistry, "extender.service.sdk.get.duration", System.currentTimeMillis() - methodStart, "sdk", hash);
-
-        return new File(sdkDirectory, "defoldsdk");
+                MetricsWriter.metricsCounterIncrement(meterRegistry, "extender.service.sdk.get.download", "sdk", hash);
+            }
+            MetricsWriter.metricsTimer(meterRegistry, "extender.service.sdk.get.duration", System.currentTimeMillis() - methodStart, "sdk", hash);
+            return sdk;
+        });
     }
 
-    public File getLocalSdk() {
+    public DefoldSdk getLocalSdk() {
         LOGGER.info("Using local Defold SDK at {}", dynamoHome.toString());
-        return dynamoHome;
+        return new DefoldSdk(dynamoHome, LOCAL_VERSION, this);
     }
 
     public boolean isLocalSdkSupported() {
         return dynamoHome != null;
     }
 
+    protected void evictCache() {
+        synchronized (cacheReferenceCount) {
+            try {
+                LOGGER.info("Cache eviction called");
+                // Delete old SDK:s
+                Comparator<Path> refCountComparator = Comparator.comparing(path -> getSdkRefCount(path.getFileName().toString()));
+
+                Files
+                        .list(baseSdkDirectory)
+                        .filter(path -> !path.getFileName().toString().startsWith("tmp")
+                                    && !path.toString().endsWith(".delete")
+                                    && !path.getFileName().toString().equals(TEST_SDK_DIRECTORY))
+                        .sorted(refCountComparator.reversed())
+                        .skip(cacheSize)
+                        .forEach(this::deleteCachedSdk);
+            } catch (IOException exc) {
+                LOGGER.error("Error during cache eviction", exc);
+            }
+        }
+    }
+
     private void deleteCachedSdk(Path path) {
+        String sdkHash = path.getFileName().toString();
+        if (getSdkRefCount(sdkHash) != 0) {
```

Comment:
Still check if ref count !=0 because sdk can marked as out of cache but it can be used by any build job.